src/Nethermind/Nethermind.Serialization.Rlp/MyTxDecoder/BlobTxDecoder.cs

```diff
@@ -0,0 +1,357 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
+using Nethermind.Serialization.Rlp.Eip2930;
+
+namespace Nethermind.Serialization.Rlp.MyTxDecoder;
+
+public sealed class BlobTxDecoder(bool lazyHash = true, Func<Transaction>? transactionFactory = null) : ITxDecoder
+{
+    private static readonly AccessListDecoder AccessListDecoder = new();
+    private readonly bool _lazyHash = lazyHash;
+    private readonly Func<Transaction> _createTransaction = transactionFactory ?? (() => new Transaction());
+
+    public Transaction? Decode(Span<byte> transactionSequence, RlpStream rlpStream, RlpBehaviors rlpBehaviors)
+    {
+        Transaction transaction = _createTransaction();
+        transaction.Type = TxType.Blob;
+
+        int positionAfterNetworkWrapper = 0;
+        if (rlpBehaviors.HasFlag(RlpBehaviors.InMempoolForm))
+        {
+            int networkWrapperLength = rlpStream.ReadSequenceLength();
+            positionAfterNetworkWrapper = rlpStream.Position + networkWrapperLength;
+            int rlpLength = rlpStream.PeekNextRlpLength();
+            transactionSequence = rlpStream.Peek(rlpLength);
+        }
+
+        int transactionLength = rlpStream.ReadSequenceLength();
+        int lastCheck = rlpStream.Position + transactionLength;
+
+        DecodePayload(transaction, rlpStream, rlpBehaviors);
+
+        if (rlpStream.Position < lastCheck)
+        {
+            DecodeSignature(transaction, rlpStream, rlpBehaviors);
+        }
+
+        if ((rlpBehaviors & RlpBehaviors.AllowExtraBytes) == 0)
+        {
+            rlpStream.Check(lastCheck);
+        }
+
+        if (rlpBehaviors.HasFlag(RlpBehaviors.InMempoolForm))
+        {
+            DecodeShardBlobNetworkWrapper(transaction, rlpStream);
+
+            if ((rlpBehaviors & RlpBehaviors.AllowExtraBytes) == 0)
+            {
+                rlpStream.Check(positionAfterNetworkWrapper);
+            }
+
+            if ((rlpBehaviors & RlpBehaviors.ExcludeHashes) == 0)
+            {
+                transaction.Hash = CalculateHashForNetworkPayloadForm(transactionSequence);
+            }
+        }
+        else if ((rlpBehaviors & RlpBehaviors.ExcludeHashes) == 0)
+        {
+            if (_lazyHash && transactionSequence.Length <= ITxDecoder.MaxDelayedHashTxnSize)
+            {
+                // Delay hash generation, as may be filtered as having too low gas etc
+                transaction.SetPreHashNoLock(transactionSequence);
+            }
+            else
+            {
+                // Just calculate the Hash as txn too large
+                transaction.Hash = Keccak.Compute(transactionSequence);
+            }
+        }
+
+        return transaction;
```

Comment:
I would appreciate a small note on this `ShardBlobNetworkWrapper` component since it makes this logic far more complicated than in the other cases.